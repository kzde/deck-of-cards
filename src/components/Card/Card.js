import React from 'react';
import Paper from '@material-ui/core/Paper';
import suits from '../../constant/index';
import './Card.scss';

const Card = (props) => {
    const { card } = props;
    return (
        <div className="Card">
            <Paper>
                <div>{suits[card.group]}</div>
                <div className={card.group}>{card.value}</div>
            </Paper>
        </div>
    )
}

export default Card;