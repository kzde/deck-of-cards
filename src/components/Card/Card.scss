.Card {
    width: 120px;
    margin: 20px;

    div {
        display: flex;
        flex-direction: column;
        align-items: center;
        padding: 20px;
        font-size: 20px;
    }

    .heart,
    .diamond {
        color: red;
    }
}