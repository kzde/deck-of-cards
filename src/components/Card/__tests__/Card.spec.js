import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../Card';
jest.mock('@material-ui/core/Paper', () => 'Paper');

const props = {
    card: {
        group: 'club',
        value: 'queen'
    }
}
describe('Card', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(<Card {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})