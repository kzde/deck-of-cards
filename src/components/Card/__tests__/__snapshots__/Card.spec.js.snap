// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should renders correctly 1`] = `
<div
  className="Card"
>
  <Paper>
    <div>
      ♣️
    </div>
    <div
      className="club"
    >
      queen
    </div>
  </Paper>
</div>
`;
